R/functions_netCDF.R

```diff
@@ -230,268 +512,307 @@ create_netCDF_from_raster_with_variables <- function(x, time_bounds,
+#'     is set to \code{FALSE} then the Z axis will be the third dimension.
+#'     If \var{has_T_timeAxis} is set to \code{TRUE} then the Z axis will be the
+#'     fourth dimension.
+#' @param isGridded A logical value. Represents whether the location data is on
+#'   a regular grid or not.
+#' @param grid filename (character). File containing the grid information
+#'   (i.e. resolution, extent) of the data. Supported file types are the 'native'
+#'   raster package format and those that can be read via rgdal.
+#' @param locations A SpatialPoints object or a matrix or data.frame with two
+#'   columns containing long and lat values. Data must be organized by long, lat.
+#' @param crs character or object of class 'CRS'.
+#' @param force_v4 A logical value. Force version 4 of netCDF.
+#'
+#' @return This function is used for the side-effect of filling a file.
+#'
+#' @examples
+#' #############################################################################
+#' # example - create an empty netcdf with a three dimensions (lat, long, time)
+#' #############################################################################
+#'
+#' # create dummy data ---------------------------------------------------------
+#'
+#' someData <- rnorm(100, 7, 30)
+#' data <- array(someData, c(10, 10))
+#'
+#' locations <- data.frame(X_WGS84 = c(rep(-124.5938, 5), rep(-124.5312, 5)),
+#'                         Y_WGS84 = rep(c(47.90625, 47.96875, 48.03125, 48.09375,
+#'                                    48.15625), 2))
+#'
+#' crs <- '+init=epsg:4326'
+#'
+#' annual_time_bounds <- c(43737, 44102, 44102, 44467, 44467, 44832, 44832, 45197,
+#'                         45197, 45563, 45563, 45928, 45928, 46293, 46293, 46658,
+#'                         46658, 47024, 47024, 47389) # beginning and end of year
+#'
+#' outFileName <- 'dummynetcdf.nc'
+#'
+#' # define attributes ---------------------------------------------------------
+#'
+#' # time attribute
+#' time_attributes <- list(
+#'    name = 'time',
+#'    units = 'days since 1900-01-01',
+#'    calendar = 'standard',
+#'    unlim = TRUE,
+#'    vals = c(43554, 43920, 44285, 44650, 45015, 45381, 45746, 46111, 46476, 46842)# mid point of year
+#'    )
+#'
+#' # variable attributes
+#' var_attributes <- list(
+#'    name = 'JulyTemp',
+#'    long_name = 'Annual Mean July Temperature',
+#'    units = 'Celsius',
+#'    description = 'example data!'
+#'    )
+#'
+#' # global attributes
+#' global_attributes <- list(
+#'    title = "",
+#'    institution = 'Southwest Biological Science Center, U.S. Geological Survey',
+#'    description = 'how this data was made',
+#'    source = paste(
+#'      "SOILWAT2 (v4.2.0);",
+#'      "rSOILWAT2 (v2.3.2);",
+#'      "rSFSW2 (v3.1.2)"
+#'    ),
+#'    source_id = "SOILWAT2",
+#'    realm = "land",
+#'    parent_mip_era = "CMIP5",
+#'    parent_experiment_id = "RCP45",
+#'    parent_source = "CanESM2",
+#'    parent_variant_label = "r1i1p1f1",
```

Comment:
CMIP5 used rip as variant label, i.e., it didn't include information about the forcing (index) in the variant label -- CMIP6 uses ripf as it added the forcing index

Reply:
added

Reply:
lines 801 and 804 continue to be inconsistent

Reply:
R/functions_netCDF.R, line 802 (`parent_variant_label = "r1i1p1f1"`) is still in conflict with line 799 (parent_mip_era = "CMIP5").

Reply:
fixed